esp-hal/src/spi/master.rs

```diff
@@ -3123,12 +3135,25 @@ impl Driver {
         no_mosi_miso: bool,
         data_mode: DataMode,
     ) -> Result<(), Error> {
+        let three_wire = cmd.mode() == DataMode::SingleThreeWire
+            || address.mode() == DataMode::SingleThreeWire
+            || data_mode == DataMode::SingleThreeWire;
+
+        if three_wire
+            && ((cmd != Command::None && cmd.mode() != DataMode::SingleThreeWire)
+                || (address != Address::None && address.mode() != DataMode::SingleThreeWire)
+                || data_mode != DataMode::SingleThreeWire)
+        {
+            return Err(Error::ArgumentsInvalid);
```

Comment:
Should we use Unsupported instead?

Reply:
I was considering that (and actually I used it first) but the description suggests something else (well I could change the description ofc) - but on the other hand looking at how it's currently used it might be fine - I don't have a preference - happy to change it

Reply:
really no preference on my side - maybe there are more opinions on that - if not I will remove `ArgumentsInvalid`

(maybe rename `Unsupported` to `OperationUnsupported`?)